src/deploy/functions/runtimes/discovery/v1alpha1.ts

```diff
@@ -180,59 +175,71 @@ function assertManifestEndpoint(ep: ManifestEndpoint, id: string): void {
   if (triggerCount > 1) {
     throw new FirebaseError("Multiple triggers defined for endpoint" + id);
   }
-  if (backend.isEventTriggered(ep)) {
+  if (build.isEventTriggered(ep)) {
     requireKeys(prefix + ".eventTrigger", ep.eventTrigger, "eventType", "eventFilters");
     assertKeyTypes(prefix + ".eventTrigger", ep.eventTrigger, {
       eventFilters: "object",
       eventFilterPathPatterns: "object",
       eventType: "string",
-      retry: "boolean",
-      region: "string",
+      retry: "Field<boolean>",
+      region: "Field<string>",
       serviceAccount: "string?",
       serviceAccountEmail: "string?",
       channel: "string",
     });
-  } else if (backend.isHttpsTriggered(ep)) {
+  } else if (build.isHttpsTriggered(ep)) {
     assertKeyTypes(prefix + ".httpsTrigger", ep.httpsTrigger, {
       invoker: "array?",
     });
-  } else if (backend.isCallableTriggered(ep)) {
+  } else if (build.isCallableTriggered(ep)) {
     // no-op
-  } else if (backend.isScheduleTriggered(ep)) {
+  } else if (build.isScheduleTriggered(ep)) {
     assertKeyTypes(prefix + ".scheduleTrigger", ep.scheduleTrigger, {
-      schedule: "string",
-      timeZone: "string?",
+      schedule: "Field<string>",
+      timeZone: "Field<string>?",
       retryConfig: "object?",
     });
-    assertKeyTypes(prefix + ".scheduleTrigger.retryConfig", ep.scheduleTrigger.retryConfig || {}, {
-      retryCount: "number?",
-      maxDoublings: "number?",
-      minBackoffSeconds: "number?",
-      maxBackoffSeconds: "number?",
-      maxRetrySeconds: "number?",
-    });
-  } else if (backend.isTaskQueueTriggered(ep)) {
+    if (ep.scheduleTrigger.retryConfig) {
+      assertKeyTypes(
+        prefix + ".scheduleTrigger.retryConfig",
+        ep.scheduleTrigger.retryConfig,
+        {
+          retryCount: "Field<number>?",
+          maxDoublings: "Field<number>?",
+          minBackoffSeconds: "Field<number>?",
+          maxBackoffSeconds: "Field<number>?",
+          maxRetrySeconds: "Field<number>?",
+          maxRetryDuration: "Field<string>?",
+          minBackoffDuration: "Field<string>?",
+          maxBackoffDuration: "Field<string>?",
+        },
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        (k: string, _v: any) =>
```

Comment:
I don't get what this does. It sounds like it allows for any value in the Duration functions? We could hypothetically say that "Duration" fields never supported params and so we should just convert them using durationToSeconds.

Reply:
That _is_ much cleaner. Done.